torch/csrc/jit/codegen/cuda/ir_utils.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/type.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+class Statement;
+
+namespace ir_utils {
+
+template <
+    typename FilterType,
+    template <typename, typename...> typename Container,
```

Comment:
idea: instead of trying to fit the container type, how about we pass it more generically and use std::begin(c) ... std::end(c) to iterate it? (that's what range based for does anyway) this way it would work even with plain C arrays.

https://en.cppreference.com/w/cpp/iterator/begin

Reply:
I'd love to know what that would look like. Can you show some sketch?